src/sentry/incidents/logic.py

```diff
@@ -1289,18 +1299,13 @@ def get_alert_rule_trigger_action_pagerduty_service(
 
 
 def get_alert_rule_trigger_action_sentry_app(organization, sentry_app_id):
-    try:
-        # query for the sentry app but make sure it's installed on that org
-        sentry_app = SentryApp.objects.get(
-            installations__organization_id=organization.id,
-            installations__status=SentryAppInstallationStatus.INSTALLED,
-            installations__date_deleted=None,
-            id=sentry_app_id,
-        )
-    except SentryApp.DoesNotExist:
-        raise InvalidTriggerActionError("No SentryApp found.")
+    from sentry.services.hybrid_cloud.app import app_service
+
+    for installation in app_service.get_installed_for_organization(organization_id=organization.id):
+        if installation.sentry_app.id == sentry_app_id:
```

Comment:
Should we have a method or filter parameter for org + sentry app?

Reply:
I totally, think so :)  But this was safe and easy to build on what existed today.  The problem is, I didn't want to move this over to `get_many` as a general set of filters, because actually the logic of "what is `get_installed_for_organization`" is kind of itself a domain encapsulation.  When I tried using several explicit filters, it was very easy to get wrong, very hard to understand why, and it didn't better represent the original encapsulation.

In the end, I think this is better than too much generalization.